pages/new.tsx

```diff
@@ -27,7 +29,7 @@ export default function NewFilePage() {
   const { userData, firebaseUser } = useNullableUserContext();
   const router = useRouter();
   const [lang, setLang] = useState<Language>('cpp');
-  const [fileName, setFileName] = useState('');
+  const [fileName, setFileName] = useState(() => generateRandomFileName());
   const [defaultPerimssion, setDefaultPermission] = useState<
```

Comment:
I just realized -- this is actually really stupid, but I think you might have to move this into a `useEffect()`, otherwise you might get a hydration mismatch error. This is because Next.js will render this page once on the server, and the client will render it again, and they'll generate two different names with two different seeds, causing the hydration mismatch. So the fix would be something like

```
  const [fileName, setFileName] = useState('');
  useEffect(() => setFileName(generateRandomFileName()), []);
```

And the effect would only run on the client, once.

Reply:
Oops, I will fix this right now.